Locker/Locker/Locker.swift

```diff
@@ -49,19 +49,26 @@ public class Locker: NSObject {
 
     // MARK: - Handle secrets (store, delete, fetch)
 
-    @objc(setSecret:forUniqueIdentifier:error:)
-    public static func setSecret(_ secret: String, for uniqueIdentifier: String) throws {
+    public static func setSecret(_ secret: String, for uniqueIdentifier: String, completion: ((LockerError?) -> Void)? = nil) {
     #if targetEnvironment(simulator)
         Locker.userDefaults?.set(secret, forKey: uniqueIdentifier)
     #else
-        setSecretForDevice(secret, for: uniqueIdentifier) { error in
-            guard let error = error else {
-                return
-            }
-            throw error
-        }
+        setSecretForDevice(secret, for: uniqueIdentifier, completion: { error in
+            completion?(error)
+        })
     #endif
+    }
 
+    @available(swift, obsoleted: 1.0)
+    @objc(setSecret:forUniqueIdentifier:completion:)
+    static func setSecret(_ secret: String, for uniqueIdentifier: String, completion: ((NSError?) -> Void)? = nil) {
```

Comment:
How does `completed` sound to you? 🤔  just thinking out loud which word is better :)

Reply:
actually "completed" is a bit more descriptive than "completion". Nice one 😄